trestle/core/control_interface.py

```diff
@@ -1091,14 +1091,33 @@ def insert_imp_req_into_component(
             have the same source and specify the same control
         """
         for control_imp in as_list(component.control_implementations):
+            _, control_imp_param_dict, _ = ControlInterface.get_rules_and_params_dict_from_item(control_imp)
+            control_imp_rule_param_ids = [d['name'] for d in control_imp_param_dict.values()]
             if profile_title != ModelUtils.get_title_from_model_uri(trestle_root, control_imp.source):
                 continue
             for imp_req in as_list(control_imp.implemented_requirements):
                 if imp_req.control_id == new_imp_req.control_id:
+                    _, imp_req_param_dict, _ = ControlInterface.get_rules_and_params_dict_from_item(imp_req)
+                    imp_req_rule_param_ids = [d['name'] for d in imp_req_param_dict]
                     status = ControlInterface.get_status_from_props(new_imp_req)
                     ControlInterface.insert_status_in_props(imp_req, status)
                     imp_req.description = new_imp_req.description
                     statement_dict = {stat.statement_id: stat for stat in as_list(imp_req.statements)}
+                    # update set parameter values with values from markdown - but only for rule param vals
+                    for set_param in as_list(new_imp_req.set_parameters):
+                        if set_param.param_id in (control_imp_rule_param_ids + imp_req_rule_param_ids):
```

Comment:
rather a stylistic comment but Sonar would agree, we can use "guard clauses" to make code cleaner. Instead of doing something when condition is met, we can skip the step if condition is not met
```
if set_param.param_id not in ...
    continue
    
# Code goes here
```

Reply:
This could be another thing to mention to Alejandro - and also convey how you review code and provide feedback.  I can address it Thurs if he is not able to.